src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp

```diff
@@ -16,7 +16,11 @@
 #include <string>
 #include <string.h>
 #include <sys/stat.h>
-#ifdef HAVE_SYS_SYSCTL_H
```

Comment:
Looking into `CMakeErrors.log` instead, I found some more details, which explains the problem:

````
Dermining if files sys/sysctl.h exist failed with the following output:
Change Dir: /home/jostein/build/coreclr/bin/obj/FreeBSD.x64.Debug/CMakeFiles/CMakeTmp

Run Build Command:"/usr/local/bin/gmake" "cmTC_688ff/fast"
/usr/local/bin/gmake -f CMakeFiles/cmTC_688ff.dir/build.make CMakeFiles/cmTC_688ff.dir/build
gmake[1]: Entering directory '/home/jostein/build/coreclr/bin/obj/FreeBSD.x64.Debug/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_688ff.dir/CheckIncludeFiles.c.o
/usr/local/bin/clang37  -D_FILE_OFFSET_BITS=64 -Wall -std=c11  -I/usr/local/include    -o CMakeFiles/cmTC_688ff.dir/CheckIncludeFiles.c.o   -c /home/jostein/build/coreclr/bin/obj/FreeBSD.x64.Debug/CMakeFiles/CMakeTmp/CheckIncludeFiles.c
In file included from /home/jostein/build/coreclr/bin/obj/FreeBSD.x64.Debug/CMakeFiles/CMakeTmp/CheckIncludeFiles.c:2:
/usr/include/sys/sysctl.h:804:40: error: unknown type name 'size_t'
int     sysctl(const int *, u_int, void *, size_t *, const void *, size_t);
                                           ^
/usr/include/sys/sysctl.h:805:40: error: unknown type name 'size_t'
int     sysctlbyname(const char *, void *, size_t *, const void *, size_t);
                                           ^
/usr/include/sys/sysctl.h:806:42: error: unknown type name 'size_t'
int     sysctlnametomib(const char *, int *, size_t *);
````

Basically (as can be seen in my commit), on FreeBSD you cannot simply include `sys/sysctl.h` without the proper prerequisites from `sys/types.h`. This is what's causing the cmake-check to fail.

For now, I`d say that the changes required to make this test pass is not worth it. If we have more `sys/sysctl.h` related issued in the future, we can reconsider. Sounds OK?